localstack/services/events/provider_v2.py

```diff
@@ -578,25 +656,41 @@ def _delete_target_sender(self, ids: TargetIdList, rule) -> None:
                 except KeyError:
                     LOG.error(f"Error deleting target service {target_arn}.")
 
-    def _put_entries(self, context: RequestContext, entries: PutEventsRequestEntryList) -> list:
-        failed_entries = []
+    def _process_entries(
+        self, context: RequestContext, entries: PutEventsRequestEntryList
+    ) -> tuple[PutEventsResultEntryList, int]:
+        processed_entries = []
+        failed_entry_count = 0
         for event in entries:
             event_bus_name = event.get("EventBusName", "default")
+            if event_failed_validation := validate_event(event):
+                processed_entries.append(event_failed_validation)
+                failed_entry_count += 1
+                continue
+            event = format_event(event, context.region, context.account_id)
             store = self.get_store(context)
-            event_bus = self.get_event_bus(event_bus_name, store)
-            # TODO add pattern matching
+            try:
+                event_bus = self.get_event_bus(event_bus_name, store)
+            except ResourceNotFoundException:
+                # ignore events for non-existing event buses but add processed event
+                processed_entries.append({"EventId": event["id"]})
+                continue
             matching_rules = [rule for rule in event_bus.rules.values()]
             for rule in matching_rules:
-                for target in rule.targets.values():
-                    target_sender = self._target_sender_store[target["Arn"]]
-                    try:
-                        target_sender.send_event(event)
-                    except Exception as error:
-                        failed_entries.append(
-                            {
-                                "Entry": event,
-                                "ErrorCode": "InternalException",
-                                "ErrorMessage": str(error),
-                            }
-                        )
-        return failed_entries
+                event_pattern = rule.event_pattern
```

Comment:
super nit: `event_pattern_str` or str typing could clarify the type here (especially given we have both versions in our codebase)

Reply:
Now for the v2 provider, since we use only the Java matching engine we only ever have event pattern as a string, we don't parse it into a JSON object. In addition, the API string type is called `EventPattern`, therefor I would keep it without the str suffix. 
Also, I think the naming convention of match_rule is misleading since we don't match the rule but the rule_pattern and only pass in the pattern. We should think of updating this in the `event_ruler`.

Reply:
> Now for the v2 provider, since we use only the Java matching engine we only ever have event pattern as a string, we don't parse it into a JSON object. In addition, the API string type is called EventPattern, therefor I would keep it without the str suffix.

It's a good thing we don't need JSON parsing for the event pattern anymore in the new provider, but right one line below we have the counter-example of transforming an `event` into an `event_str` 😬 . Looking over the many implementations in LocalStack, this `str` vs `json` duality was very confusing, and being explicit can help to clarify the API usage here.

---

How do you think does `rule` differ from `rule_pattern`?
> the naming convention of match_rule is misleading since we don't match the rule but the rule_pattern and only pass in the pattern.

## Background

The terminology `Ruler.matchesRule(event, rule)`  in the `event_ruler.py` follows the official AWS [event-ruler](https://github.com/aws/event-ruler?tab=readme-ov-file#static-rule-matching) and aims to be more generic than just for EventBridge. Notice that different services use slightly different terminology:
* [Amazon EventBridge event patterns](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-patterns.html)
* [Amazon EventBridge Pipes filtering](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-event-filtering.html), also called `FilterCriteria`
* Event Source Mapping [Lambda event filtering](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html), also called `FilterCriteria`
* [Amazon SNS message filtering](https://docs.aws.amazon.com/sns/latest/dg/sns-message-filtering.html), also called `filter policy`

Reply:
json vs strg makes sense - the old fallacy if you build it you know it and think its obvious to everyone else